### ISSUE TITLE
[exporter/prometheusremotewriteexporter] Deprecate and Remove `export_created_metric` feature.

### ISSUE THREAD

**Issue by ArthurSens** [open-telemetry/opentelemetry-collector-contrib#35003]
### Component(s)

exporter/prometheusremotewrite

### Describe the issue you're reporting

Although the intention with https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/17417 was good to be compliant with OpenMetrics, it's important to mention that OpenMetrics is an data exposition format, while Prometheus Remote-Write is a data transport protocol. 

The original goal of `StartTimeUnixNano` and OM's `_created` metrics is to identify resets of monotonically increasing metrics. 

Knowing that our exporter exposes `StartTimeUnixNano` as a separate metric, and that PRWv1 potentially splits metrics into multiple requests, there's no guarantee that `_created` and its related metric will be sent in the same request, which makes the usage of this feature super super difficult and inefficient. The current implementation in Prometheus when receiving this metric will just create another timeseries and will not use it to identify resets, making this feature in the collector pointless.

My suggestion here is that we remove this feature altogether. Let's keep in mind that this is a prwexporter and not a openmetrics exporter.

**Comment by dashpole** [comment]
Since PRW v2.0 will have the created timestamp in the proto, this will become moot after we switch to that.  Should we just leave it around until then?  IIRC, it is disabled by default.  I'm not sure if people are using it or not. We can add a warning or something to the documentation if you want to make sure people know it doesn't work with Promethues' CT support

**Comment by ArthurSens** [comment]
I'd rather remove it, if possible. For PRWv2 I think we'll always do this conversion, it shouldn't be opt-out 🤔. While in PRWv1 it just doesn't work at all, so there's no reason to enable it

**Comment by dashpole** [comment]
cc @kovrus do you have any thoughts here?

**Comment by ArthurSens** [comment]
Sorry, this is still on my list. Just struggling to allocate time for it 😬

**Comment by pokgak** [comment]
@ArthurSens here's the PR for deprecation: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/36214

If I understand the guideline correctly, removing the config option itself have to be done separately in the next release right?

**Comment by ArthurSens** [comment]
> @ArthurSens here's the PR for deprecation: #36214
> 
> If I understand the guideline correctly, removing the config option itself have to be done separately in the next release right?

Yes, for complete removal it will take a few releases... but we can put the feature behind a feature gate and slowly decrease the stability stage until complete removal

**Comment by ArthurSens** [comment]
Now that v0.114.0 has been released, we can switch the feature flag to `StageBeta`. Should we re-open this issue until the whole deprecation is finished? :)

**Comment by TylerHelmuth** [comment]
Yes

**Comment by ArthurSens** [comment]
I completely forgot about this, my bad! @perebaj, do you wanna work on this? I can tackle it this week if you're busy with other things :)

**Comment by perebaj** [comment]
> I completely forgot about this, my bad! [@perebaj](https://github.com/perebaj), do you wanna work on this? I can tackle it this week if you're busy with other things :)

Yes I can, dont worry

**Comment by ArthurSens** [comment]
Fixed by #37399
